### ISSUE TITLE
REQUEST Science able to select nebulae for scanning (and setDescription available for them).

### ISSUE THREAD

**Issue by Wrongtown** [daid/EmptyEpsilon#83]
Given the whole [SpaceClouds trope](http://tvtropes.org/pmwiki/pmwiki.php/Main/SpaceClouds) I figure it's only a matter of time before some enterprising (no pun intended) scenario-writer decides they want to introduce nebula that has a helpful or harmful effect. If we allow Science to scan a nebula before Helm takes the ship in it could add another layer to scenarios. Based on the position of the ship the nebula could be made to appear to drain energy, add energy, disable systems, damage systems, play jazz music etc.

One thing to consider is that currently ships within a nebula can be selected and scanned, so if the "hitbox" for selection by science is the full size of the nebula then that could be lost. Not sure how hard it would be to check "Can the player see a ship at those co-ordinates?" and if False select the nebula instead.

So this request could potentially have a couple of layers to it.
**MVP**
- As Scenario Writer I want to be able to use setDescription for any given nebula so the Science Officer can see why weird effects are occurring in my scenarios.

**Nice To Have**
- As Science Officer I want to be able to scan a nebula and have some kind of description returned so I can feel I helped the Captain can make informed choices.
- As Scenario Writer I want to be able to use setScanningParameters for any given nebula so I can make it easier or harder to scan as plot demands.
- As a Scenario Writer I want any nebula which doesn't have an explicit description via setDescription to have a default description (ideally one of a few) which equates to "It's a normal nebula." so I don't have to manually include it for every nebula on the map.

**Comment by nallath** [comment]
Well, currently the nebulas already have a harmfull effect; you can't scan through em (or atleast, your long range stuff doesn't work)

**Comment by Wrongtown** [comment]
Yeah I've been a bit confused by the results of my testing this evening. 

So far if my Science scope extends into the nebula far enough to include the coordinates of the enemy, then I can select them and scan them. It does block Line of Sight for stuff that's **behind** the nebula but I thought ships outside shouldn't be able to see ships inside?

**Comment by daid** [comment]
https://github.com/daid/EmptyEpsilon/commit/0897c48a2ca3c6d2a86d8452990c94d2f705db31

This changes the following:
* When setDescription is called on any object, this object will be selectable in science.
* When setScanningParameters is called on any object, this object will be selectable in science, and scannable.

Now I just need to fix that you can see an object is scanned, as currently this is only readable on ships.

**Comment by daid** [comment]
Also, science is still able to see 5km around the ship, no matter the Nebula's.

**Comment by Wrongtown** [comment]
Very nice!
